agent/acs/handler/task_manifest_handler.go

```diff
@@ -0,0 +1,293 @@
+	return func(message *ecsacs.TaskStopVerificationAck) {
+		taskManifestHandler.messageBufferTaskStopVerificationAck <- message
+	}
+}
+
+func (taskManifestHandler *taskManifestHandler) start() {
+	// Task manifest and it's ack
+	go taskManifestHandler.handleTaskManifestMessage()
+	go taskManifestHandler.sendTaskManifestMessageAck()
+
+	// Task stop verification message and it's ack
+	go taskManifestHandler.sendTaskStopVerificationMessage()
+	go taskManifestHandler.handleTaskStopVerificationAck()
+
+}
+
+func (taskManifestHandler *taskManifestHandler) getMessageId() string {
+	taskManifestHandler.lock.Lock()
+	defer taskManifestHandler.lock.Unlock()
+	return taskManifestHandler.messageId
+}
+
+func (taskManifestHandler *taskManifestHandler) setMessageId(messageId string) {
+	taskManifestHandler.lock.Lock()
+	defer taskManifestHandler.lock.Unlock()
+	taskManifestHandler.messageId = messageId
+}
+
+func (taskManifestHandler *taskManifestHandler) sendTaskManifestMessageAck() {
+	for {
+		select {
+		case messageBufferTaskManifestAck := <-taskManifestHandler.messageBufferTaskManifestAck:
+			taskManifestHandler.ackTaskManifestMessage(messageBufferTaskManifestAck)
+		case <-taskManifestHandler.ctx.Done():
+			return
+		}
+	}
+}
+
+func (taskManifestHandler *taskManifestHandler) handleTaskStopVerificationAck() {
+	for {
+		select {
+		case messageBufferTaskStopVerificationAck := <-taskManifestHandler.messageBufferTaskStopVerificationAck:
+			if err := taskManifestHandler.handleSingleMessageVerificationAck(messageBufferTaskStopVerificationAck); err != nil {
+				seelog.Warnf("Error handling Verification ack with messaageID: %s, error: %v",
+					messageBufferTaskStopVerificationAck.MessageId, err)
+			}
+		case <-taskManifestHandler.ctx.Done():
+			return
+		}
+	}
+}
+
+func (taskManifestHandler *taskManifestHandler) clearAcks() {
+	for {
+		select {
+		case <-taskManifestHandler.messageBufferTaskManifestAck:
+		case <-taskManifestHandler.messageBufferTaskStopVerificationAck:
+		default:
+			return
+		}
+	}
+}
+
+func (taskManifestHandler *taskManifestHandler) ackTaskManifestMessage(messageID string) {
+	seelog.Debugf("Acking task manifest message id: %s", messageID)
+	err := taskManifestHandler.acsClient.MakeRequest(&ecsacs.AckRequest{
+		Cluster:           aws.String(taskManifestHandler.cluster),
+		ContainerInstance: aws.String(taskManifestHandler.containerInstanceArn),
+		MessageId:         aws.String(messageID),
+	})
+	if err != nil {
+		seelog.Warnf("Error 'ack'ing TaskManifestMessage with messageID: %s, error: %v", messageID, err)
+	}
+}
+
+func (taskManifestHandler *taskManifestHandler) sendTaskTerminationVerificationMessage(
+	message *ecsacs.TaskStopVerificationMessage) error {
+
+	seelog.Debugf("Sending Task Termination Verification Message: %s", message.String())
+	err := taskManifestHandler.acsClient.MakeRequest(message)
+	if err != nil {
+		seelog.Warnf("Error 'ack'ing request with messageID: %s, error: %v", message.MessageId, err)
```

Comment:
You already logged this error iin sendTaskStopVerificationMessage so i don't think you need to log it here?

Reply:
Yeah makes sense.